test/array.js

```diff
@@ -202,4 +202,19 @@ describe('Array types', () => {
       .of(itemSchema)
       .validate(value);
   });
+
+  it('should treat sparse arrays empty slots as undefined for the purpose of validation', async () => {
+    let sparseArray = new Array(2);
+    sparseArray[1] = 1;
+    let value = await array().of(number()).validate(sparseArray);
+    expect(0 in sparseArray).to.be.false()
+    expect(0 in value).to.be.false()
+    value.should.eql([undefined, 1]);
```

Comment:
```suggestion
    value.should.eql([, 1]);
```

more clearly signals that this should be sparse